hydra-cluster/src/HydraNode.hs

```diff
@@ -161,23 +162,50 @@ waitForAll tracer delay nodes expected = do
         _ ->
           tryNext c msgs stillExpected
 
+-- | Create a commit tx using the hydra-node for later submission
+externalCommit :: HydraClient -> UTxO -> IO Tx
+externalCommit HydraClient{hydraNodeId} utxos =
+  runReq defaultHttpConfig request
+    <&> responseBody
+    >>= \DraftCommitTxResponse{commitTx} -> pure commitTx
+ where
+  request =
+    Req.req
+      POST
+      (Req.http "127.0.0.1" /: "commit")
+      (ReqBodyJson (DraftCommitTxRequest{utxos} :: DraftCommitTxRequest Tx))
+      (Proxy :: Proxy (JsonResponse (DraftCommitTxResponse Tx)))
+      (Req.port $ 4000 + hydraNodeId)
+
 getMetrics :: HasCallStack => HydraClient -> IO ByteString
 getMetrics HydraClient{hydraNodeId} = do
-  response <-
-    failAfter 3 $ queryNode hydraNodeId
-  when (getResponseStatusCode response /= 200) $ failure ("Request for Hydra-node metrics failed :" <> show (getResponseBody response))
-  pure $ getResponseBody response
+  response <- failAfter 3 $ queryNode hydraNodeId
+  let respBody = Req.responseBody response
+  when (Req.responseStatusCode response /= 200) $
+    failure ("Request for Hydra-node metrics failed :" <> show respBody)
```

Comment:
This will not be called as `runReq` seemingly will throw a `HttpException` on errors.

Should: use `try` around `runReq` to provide more context.